app/Services/Search/EntitySearchService.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Services\Search;
+
+use App\Models\Entity;
+use App\Models\Attribute;
+use App\Models\QuestElement;
+use App\Models\TimelineElement;
+use App\Models\Post;
+use App\Traits\CampaignAware;
+use Meilisearch\Client;
+
+
+class EntitySearchService
+{
+    use CampaignAware;
+
+
+    /**
+     * Send search request
+     * @param string|null $query Search term
+     */
+    public function search(string $term = null): array
+    {
+        $client = new Client('http://meilisearch:7700', 'password');
+        $client->getKeys();
+        $ids = [];
+        $results = $client->index('entities')->search($term, ['filter' => 'campaign_id = ' . $this->campaign->id, 'attributesToRetrieve' => ['id', 'entity_id', 'type'], 'limit' => 20])->getHits();
+        $attributeIds = [];
+        $timelineElementIds = [];
+        $postIds = [];
+        $questElementIds = [];
+
+        foreach ($results as $result) {
```

Comment:
Split up the service into smaller functions so that I know what's going on just from the function name